### PR TITLE
PP-2672 Refactoring PaymentFinder.findByProductExternalId

### PR DESCRIPTION
**Pull request by SandorArpa**
- refactor findByProductExternalId to query publicapi for charge status if payment status is SUBMITTED
- refactor tests accordingly

### CODE REVIEW

**Comment by kakumara** [comment]
needs rebasing

File: src/main/java/uk/gov/pay/products/service/PaymentFinder.java [hunk 1]
```diff
@@ -27,34 +27,46 @@ public PaymentFinder(PaymentDao paymentDao, LinksDecorator linksDecorator, Publi
...
 
-    @Transactional
     public List<Payment> findByProductExternalId(String productExternalId) {
-        List<Payment> payments = paymentDao.findByProductExternalId(productExternalId)
+        return findPaymentEntitiesByProductExternalId(productExternalId)
                 .stream()
-                .map(paymentEntity -> linksDecorator.decorate(paymentEntity.toPayment()))
+                .map(paymentEntity -> {
+                    Payment payment = queryGovUKPaymentStatus(paymentEntity);
+                    return linksDecorator.decorate(payment);
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/products/service/PaymentFinder.java]: What happens if one of the payment throws an error here?

**Comment by kakumara** [comment]
Other than that 👍

File: src/main/java/uk/gov/pay/products/service/PaymentFinder.java (hunk 1 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/products/service/PaymentFinder.java]: Added try/catch it logs and swallows error

**Pull request merged**